packages/teams-js/test/private/externalAppCardActions.spec.ts

```diff
@@ -142,7 +144,7 @@ describe('externalAppCardActions', () => {
           const message = utils.findMessageByFunc('externalAppCardActions.processActionOpenUrl');
           if (message && message.args) {
             expect(message).not.toBeNull();
-            expect(message.args).toEqual([testAppId, testUrl.href]);
+            expect(message.args).toEqual([testAppId, testUrl.href, null]);
```

Comment:
Why is this `null` and not `undefined`?

Reply:
I spent a lot of time here, but could not find where exactly the undefined is changed to null. I think it is happening somewhere in /test/utils.ts. The type of fromElement is undefined everywhere in communication file but when I look it up from utils.findMessageByFunc, the arg come as null instead of undefined. 

But from teams-js to hubsdk 'undefined' is sent. I think that matters more. If I find more time I will investigate more

Reply:
Yeah I agree, as long as what we send to host matches what we expect there it should be fine

Reply:
I found that in communication file the message is stringified before sending across the wire. 
 ```
    if (Communication.currentWindow && Communication.currentWindow.nativeInterface) {
      logger('Sending message %i to ${targetWindowName} via framelessPostMessage interface', request.id);
      (Communication.currentWindow as ExtendedWindow).nativeInterface.framelessPostMessage(JSON.stringify(request));
    }
```

When we stringify a undefined value in an array, it is converted to null
![image](https://github.com/OfficeDev/microsoft-teams-library-js/assets/88564543/f10f5c98-be19-41c2-a15c-963c14a5f104)

on host sdk we still receive undefined - that is still mystery to me